mavros/src/plugins/waypoint.cpp

```diff
@@ -702,39 +728,86 @@ class WaypointPlugin : public plugin::PluginBase {
 	}
 
 	bool push_cb(mavros_msgs::WaypointPush::Request &req,
-			mavros_msgs::WaypointPush::Response &res)
+				mavros_msgs::WaypointPush::Response &res)
 	{
 		unique_lock lock(mutex);
 
 		if (wp_state != WP::IDLE)
 			// Wrong initial state, other operation in progress?
 			return false;
 
-		wp_state = WP::TXLIST;
+		if (req.start_index)
+		{
+			//Partial Waypoint update
+
+			if (!enable_partial_push)
+			{
+				ROS_WARN_NAMED("wp","WP: Partial Push not enabled. (Only supported on APM)");
+				res.success = false;
+				res.wp_transfered = 0;
+				return true;
+			}
+
+			if (waypoints.size() < req.start_index + req.waypoints.size())
+			{
+				ROS_WARN_NAMED("wp","WP: Partial push out of range rejected.");
+				res.success = false;
+				res.wp_transfered = 0;
+				return true;
+			}
```

Comment:
Partial update does not support extension of existing mission?

Reply:
No, it doesn't the waypoint count cant change